core/trie/bitarray.go

```diff
@@ -0,0 +1,458 @@
+package trie
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"math/bits"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+const (
+	maxUint64 = uint64(math.MaxUint64) // 0xFFFFFFFFFFFFFFFF
+	bits8     = 8
+)
+
+var emptyBitArray = new(BitArray)
```

Comment:
I think the heap allocation is fine for this since it's intended to be shared, immutable and zero instance used in different places. I don't think we need to go for duplications.